### DISCUSSION TITLE
Access podman from another wsl distro?

### DISCUSSION THREAD

**Discussion by kunxi** [containers/podman#15416] [Q&A] [answered]
I managed to install podman by following the [Windows Installation Guide](https://github.com/containers/podman/blob/main/docs/tutorials/podman-for-windows.md). And I believe podman uses the wsl2 backend with the `podman-machine-default`:

```
PS C:\Users> wsl -l
Windows Subsystem for Linux Distributions:
Debian (Default)
Ubuntu
docker-desktop
docker-desktop-data
podman-machine-default
```

My main daily drive is the Debian distro, and with docker desktop wsl integration, I can run `docker`, and `docker-compose` inside the WSL Debian distro. I wonder how could I do the similar things with podman? More concretely:

1. If I install podman in Debian, should I configure the podman remote access? Like access `podman-machine-default` from Debian?
2. If I want to remove docker desktop, I think I can alias docker to podman, but how to do the `docker-compose`? Will I need both podman-docker and docker-compose pacakge?

Thanks.

**Comment by baude** [comment, marked as answer]
On question 1, yes.  There is a lengthy list of things that need to be done to make that work; which is why we provide you with a pre-configured instance that does everything for you.  Does the distro even matter at that point?

On question 2, poke around as the answer to that is there.  In fact, when you `podman machine start` there is a helpful message about the sockets.  The following writeup may help -> https://hackernoon.com/how-to-run-podman-and-docker-compose-on-windows

**Reply by kunxi** [reply]
Thanks for the quick response. That explains why there are needs for remote access.

1. The distro does not matter from the container point of view. I just don't use powershell, and in favor of the linux environment, and there are lots of hoops jumped to make nixpkgs and home-manager working in Debian, do not want to switch to a new env.
2. I think the solution is to install docker, docker-compose in the Debian distro, and configure the podman socket as the drop-in replacement for docker.

**Reply by jfmatth** [reply]
> On question 1, yes. There is a lengthy list of things that need to be done to make that work; which is why we provide you with a pre-configured instance that does everything for you. Does the distro even matter at that point?
> 

So I see the message ```API forwarding listening on: npipe:////./pipe/docker_engine```, so does that mean another WSL instance will just "see" podman as Docker?
